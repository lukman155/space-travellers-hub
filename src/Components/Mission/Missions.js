import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission } from '../../Redux/Mission/mission';
import './mission.css';

const Mission = ({
  name, description, index, join, keyValue,
}) => {
  const dispatch = useDispatch();
  const odds = index % 2;
  const styles = {
    backgroundColor: odds ? 'rgb(182, 180, 180, 0.5)' : '#ffffff',
  };
  return (
    <div className="mission">
      <div className="mission-details" style={styles}>
        <div className="mission-name">{name}</div>
        <div className="mission-description">{description}</div>
        <div className="mission-status-container">
          {join
            ? (<span className="mission-status-Active">Active Member</span>)
            : (<span className="mission-status">NOT A MEMBER</span>) }
        </div>
        <div className="mission-button">
          {join
            ? (
              <button type="button" className="leaveMission" onClick={() => dispatch(joinMission(keyValue))}>
                Leave Mission
              </button>
            )
            : (
              <button type="button" className="joinMission" onClick={() => dispatch(joinMission(keyValue))}>
                Join Mission
              </button>
            )}
        </div>
      </div>
    </div>
  );
};

Mission.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  join: PropTypes.bool.isRequired,
  keyValue: PropTypes.string.isRequired,
};
export default Mission;
