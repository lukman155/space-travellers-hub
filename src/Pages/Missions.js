import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Mission from '../Components/Mission/Missions';
import './mission.css';
import { getMissions } from '../Redux/Mission/mission';

const Missions = () => {
  const dispatch = useDispatch();
  const missionList = useSelector((state) => state.missions);
  useEffect(() => {
    if (missionList.length) {
      return;
    }
    dispatch(getMissions());
  }, [dispatch, missionList.length]);
  return (
    <div className="mission-container">
      <ul className="mission-heading">
        <li>Mission</li>
        <li className="description">Description</li>
        <li>Status</li>
        <li />
      </ul>
      <div className="missionList">
        {missionList.map((mission, index) => (

          <Mission
            index={index}
            key={mission.mission_id}
            name={mission.mission_name}
            description={mission.description}
            join={mission.join}
            keyValue={mission.mission_id}
          />
        ))}
      </div>
    </div>
  );
};

export default Missions;
