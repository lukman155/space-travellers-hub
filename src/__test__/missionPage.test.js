import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Missions from '../Pages/Missions';
import missionsReducer from '../Redux/Mission/mission';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
  },
});

describe('Missions', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Missions />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
