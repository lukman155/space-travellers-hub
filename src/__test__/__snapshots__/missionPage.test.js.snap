// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders correctly 1`] = `
<div
  className="mission-container"
>
  <ul
    className="mission-heading"
  >
    <li>
      Mission
    </li>
    <li
      className="description"
    >
      Description
    </li>
    <li>
      Status
    </li>
    <li />
  </ul>
  <div
    className="missionList"
  />
</div>
`;
